provider/CONTRIBUTE.md

```diff
@@ -135,6 +136,120 @@ Once done, you can just press F5 and the debug will start! You can place breakpo
 
 *Note: the first time your start the debug, it can take a while, you need to be patient :-)*
 
+## How to contribute to the Azure RM provider
+
+*Note: we also invite you to read the official documentation about developing a provider [here](https://www.terraform.io/docs/plugins/basics.html#developing-a-plugin)*
+
+First you need to pick [an issue on the provider](https://github.com/terraform-providers/terraform-provider-azurerm/issues) by commenting the issue and saying that you're going to work on that, to make sure that the repo maintainers are aware that you are going to work on this issue.
+We also strongly advise that you describe the work you are planning to do, like explaining your implementation, submitting the TF schema for new resources, to make sure the discussion start early as possible with the reviewers.
+
+*Note: if there is no issue for the problem you are trying to solve, you can create one.*
+
+If you are a new contributor, there is a [good first issue](https://github.com/terraform-providers/terraform-provider-azurerm/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) label that may help you to filter the issue and start with a simple one.
+
+For each piece of code that you write into the provider, you need to make sure that you have:
+
+- the implementation in both `data source` and `resource` definition
+- acceptance test created or updated
+- documentation created or updated
+- example created or updated

```

Comment:
Example is optional. Typically for my contributions, I do not add examples.

Reply:
Done.